File: Crawlers/FMRReport.py
# Basic class to represent a FMR Crawler Report
# This class is designed to be implemented by the FMRCrawler to create list of report objects that can be exported to XML format.
class FMRCrawlerReport:

    #create a report with all relevant data attributes
    def __init__(self,fiscal_year,state_name,state_code,county_name,fips_code,hud_geo_id,msa_code,area_type,hud_region_code,zip_code,num_bedrooms,fair_market_rent,percentile_type,bedroom_dist_source,survey_source_year,adjustment_factor,
                 is_small_area_fmr,median_household_income,source_url,scrape_date,version_hash,update_freq,crawler_run_data):
        
        self._fiscal_year = fiscal_year
        self._state_name = state_name
        self._state_code = state_code
        self._county_name = county_name
        self._fips_code = fips_code
        self._hud_geo_id = hud_geo_id
        self._msa_code = msa_code
        self._area_type = area_type
        self._hud_region_code = hud_region_code
        self._zip_code = zip_code
        self._num_bedrooms = num_bedrooms
        self._fair_market_rent = fair_market_rent
        self._percentile_type = percentile_type
        self._bedroom_dist_source = bedroom_dist_source
        self._survey_source_year = survey_source_year
        self._adjustment_factor = adjustment_factor
        self._is_small_area_fmr = is_small_area_fmr
        self._median_household_income = median_household_income
        self._source_url = source_url
        self._scrape_date = scrape_date
        self._version_hash = version_hash
        self._update_freq = update_freq
        self._crawler_run_data = crawler_run_data

    #getter methods for all data attributes
    def get_fiscal_year(self):
        return self._fiscal_year
    def get_state_name(self):
        return self._state_name
    def get_state_code(self):
        return self._state_code
    def get_county_name(self):
        return self._county_name
    def get_fips_code(self):
        return self._fips_code
    def get_hud_geo_id(self):
        return self._hud_geo_id
    def get_msa_code(self):
        return self._msa_code
    def get_area_type(self):
        return self._area_type
    def get_hud_region_code(self):
        return self._hud_region_code
    def get_zip_code(self):
        return self._zip_code
    def get_num_bedrooms(self):
        return self._num_bedrooms
    def get_fair_market_rent(self):
        return self._fair_market_rent
    def get_percentile_type(self):
        return self._percentile_type
    def get_bedroom_dist_source(self):
        return self._bedroom_dist_source
    def get_survey_source_year(self):
        return self._survey_source_year
    def get_adjustment_factor(self):
        return self._adjustment_factor
    def get_is_small_area_fmr(self):
        return self._is_small_area_fmr
    def get_median_household_income(self):
        return self._median_household_income
    def get_source_url(self):
        return self._source_url
    def get_scrape_date(self):
        return self._scrape_date
    def get_version_hash(self):
        return self._version_hash
    def get_update_freq(self):
        return self._update_freq
    def get_crawler_run_data(self):
        return self._crawler_run_data

File: JavaFXGUIDemo/FMRGUI/src/main/java/crawlers/reportgui/FMRReport.java
package crawlers.reportgui;

// FMRReport class encapsulates Fair Market Rent (FMR) report details.
public class FMRReport {
    private String reportID;
    private String reportType;
    private String fiscalYear;
    private String stateName;
    private String stateCode;
    private String countyName;
    private String fipsCode;
    private String hudGeoId;
    private String msaCode;
    private String areaType;
    private String hudRegionCode;
    private String zipCode;
    private String numBedrooms;
    private String fairMarketRent;
    private String percentileType;
    private String bedroomDistSource;
    private String surveySourceYear;
    private String adjustmentFactor;
    private String isSmallAreaFmr;
    private String medianHouseholdIncome;
    private String sourceUrl;
    private String scrapeDate;
    private String versionHash;
    private String updateFreq;
    private String crawlerRunData;

    FMRReport(){}

    // Getters
    public String getReportID() {
        return reportID;
    }

    public String getReportType() {
        return reportType;
    }

    public String getCrawlerRunData() {
        return crawlerRunData;
    }

    public String getSmallAreaFmr() {
        return isSmallAreaFmr;
    }

    public String getAdjustmentFactor() {
        return adjustmentFactor;
    }

    public String getFairMarketRent() {
        return fairMarketRent;
    }

    public String getMedianHouseholdIncome() {
        return medianHouseholdIncome;
    }

    public String getNumBedrooms() {
        return numBedrooms;
    }

    public String getSurveySourceYear() {
        return surveySourceYear;
    }

    public String getAreaType() {
        return areaType;
    }

    public String getBedroomDistSource() {
        return bedroomDistSource;
    }

    public String getCountyName() {
        return countyName;
    }

    public String getFipsCode() {
        return fipsCode;
    }

    public String getFiscalYear() {
        return fiscalYear;
    }

    public String getHudGeoId() {
        return hudGeoId;
    }

    public String getHudRegionCode() {
        return hudRegionCode;
    }

    public String getMsaCode() {
        return msaCode;
    }

    public String getPercentileType() {
        return percentileType;
    }

    public String getScrapeDate() {
        return scrapeDate;
    }

    public String getSourceUrl() {
        return sourceUrl;
    }

    public String getStateCode() {
        return stateCode;
    }

    public String getStateName() {
        return stateName;
    }

    public String getUpdateFreq() {
        return updateFreq;
    }

    public String getVersionHash() {
        return versionHash;
    }

    public String getZipCode() {
        return zipCode;
    }

    // Setters
    public void setReportID(String reportID) {
        this.reportID = reportID;
    }

    public void setReportType(String reportType) {
        this.reportType = reportType;
    }

    public void setFiscalYear(String fiscalYear) {
        this.fiscalYear = fiscalYear;
    }

    public void setStateName(String stateName) {
        this.stateName = stateName;
    }

    public void setStateCode(String stateCode) {
        this.stateCode = stateCode;
    }

    public void setCountyName(String countyName) {
        this.countyName = countyName;
    }

    public void setFipsCode(String fipsCode) {
        this.fipsCode = fipsCode;
    }

    public void setHudGeoId(String hudGeoId) {
        this.hudGeoId = hudGeoId;
    }

    public void setMsaCode(String msaCode) {
        this.msaCode = msaCode;
    }

    public void setSmallAreaFmr(String isSmallAreaFmr) {
        this.isSmallAreaFmr = isSmallAreaFmr;
    }

    public void setAreaType(String areaType) {
        this.areaType = areaType;
    }

    public void setHudRegionCode(String hudRegionCode) {
        this.hudRegionCode = hudRegionCode;
    }

    public void setZipCode(String zipCode) {
        this.zipCode = zipCode;
    }

    public void setNumBedrooms(String numBedrooms) {
        this.numBedrooms = numBedrooms;
    }

    public void setFairMarketRent(String fairMarketRent) {
        this.fairMarketRent = fairMarketRent;
    }

    public void setPercentileType(String percentileType) {
        this.percentileType = percentileType;
    }

    public void setBedroomDistSource(String bedroomDistSource) {
        this.bedroomDistSource = bedroomDistSource;
    }

    public void setSurveySourceYear(String surveySourceYear) {
        this.surveySourceYear = surveySourceYear;
    }

    public void setAdjustmentFactor(String adjustmentFactor) {
        this.adjustmentFactor = adjustmentFactor;
    }

    public void setMedianHouseholdIncome(String medianHouseholdIncome) {
        this.medianHouseholdIncome = medianHouseholdIncome;
    }

    public void setSourceUrl(String sourceUrl) {
        this.sourceUrl = sourceUrl;
    }

    public void setScrapeDate(String scrapeDate) {
        this.scrapeDate = scrapeDate;
    }

    public void setVersionHash(String versionHash) {
        this.versionHash = versionHash;
    }

    public void setUpdateFreq(String updateFreq) {
        this.updateFreq = updateFreq;
    }

    public void setCrawlerRunData(String crawlerRunData) {
        this.crawlerRunData = crawlerRunData;
    }

    // String representation of the FMRReport
    @Override
    public String toString() {
        return String.format("FMR ID: %-4s %-6s %-4s %s %-10s",Integer.parseInt(reportID)+1,fiscalYear,stateCode,countyName,zipCode);
    }
}

